virtual-server/examples/go/main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	vsv1alpha "github.com/coreweave/virtual-server/api/v1alpha1"
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"kubevirt.io/client-go/kubecli"
+)
+
+// Ready waits until virtual server reach expected status
+func Ready(namespace, name string, c client.Client) string {
+	for {
+		vs := &vsv1alpha.VirtualServer{}
+		err := c.Get(context.Background(), client.ObjectKey{
+			Namespace: namespace,
+			Name:      name,
+		}, vs)
+		if err != nil {
+			log.Fatalf("cannot obtain virtual server status: %v\n", err)
+		}
+
+		cond := vs.GetReadyStatus()
+		if cond == nil {
+			return "Return"
+		} else if cond.Reason == "VirtualServerReady" && cond.Type == "Ready" && cond.Status == "True" {
+			fmt.Printf("Network:\n\tInternalIP: %s\n\tTCP: %s\n\tUDP: %s",
+				vs.Status.InternalIP(),
+				vs.Status.TCP(),
+				vs.Status.UDP(),
+			)
+			return "Ready"
+		} else if cond.Reason == "VirtualServerStopped" && cond.Type == "Ready" && cond.Status == "False" {
+			return "Stopped"
+		}
+		time.Sleep(2)
+	}
+}
+
+func main() {
+	name := "my-virtual-server"
+
+	// Get namespace or use default
+	namespace, envExist := os.LookupEnv("NAMESPACE")
+	if envExist == false {
+		namespace = "default"
+	}
+	// Uses the value of the KUBECONFIG environment variable as a filepath to a kube config file
+	c, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		log.Fatalf("Failed to create client\n")
+	}
+
+	vsv1alpha.AddToScheme(c.Scheme())
+
+	// prepare config for kubevirt client, you need to set env variable, KUBECONFIG=<path-to-kubeconfig>/.kubeconfig
+	kubevirtClientConfig := kubecli.DefaultClientConfig(&pflag.FlagSet{})
+
+	// get the kubevirt client, using which kubevirt resources can be managed.
+	kubevirtClient, err := kubecli.GetKubevirtClientFromClientConfig(kubevirtClientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain KubeVirt client: %v\n", err)
+	}
+
+	// Create a new VirtualServer with the name "my-virtual-server" to be deployed in the "default" namespace
```

Comment:
In the comment: not the "default" namespace in all instances

Reply:
as we talked about it already if the user does not set env var NAMESPACE, `default` namespace will be used.